### ISSUE TITLE
Improve the template code of the Finder module 

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#41304]
A new `<search>` element has been added to the HTML specification. 
see [4.4.15 The search element](https://html.spec.whatwg.org/multipage/grouping-content.html#the-search-element)
The Finder module template uses the `role=search` attribute in the `form` tag.
```
<form 
   class="mod-finder js-finder-searchform form-search" 
   action="<?php echo Route::_($route); ?>" 
   method="get" 
   role="search">
```
The incorrect use of the role=search attribute on the form element was pointed out by Adrian Roselli (see [Where to Put Your Search Role](https://adrianroselli.com/2015/08/where-to-put-your-search-role.html)

I suggest that the Finder module template should be improved:
- use the search tag
- remove the role attribute from the form element 
- if the search tag is not used (because it may not yet have good support in browsers and assistive technologies), place the role=search attribute in an additional container (see examples at Adrian Roselli)

**Comment by brianteeman** [comment]
If you read the specification and posts from others such as [scott ohara ](https://www.scottohara.me/blog/2023/03/24/search-element.html )you will see that they disagree with adrian

**Comment by brianteeman** [comment]
This is now supported by all browsers

**Comment by rdeutz** [comment]
Can someone make a PR for Joomla6?

**Comment by chmst** [comment]
Please test #45744
